tests/capi/unittest_capi_inference.cc

```diff
@@ -1437,6 +1438,175 @@ TEST (nnstreamer_capi_src, callback_invalid_param_02_n)
   EXPECT_EQ (status, ML_ERROR_NONE);
 }
 
+/**
+ * @brief Check decoded orange.png with raw data.
+ */
+static void
+check_orange_output (const ml_tensors_data_h data, const ml_tensors_info_h info, void *user_data)
+{
+  int status, fd;
+  size_t raw_data_size, data_size;
+  ssize_t data_read;
+  void *raw_data;
+  gchar *orange_raw_file;
+
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  orange_raw_file = g_build_filename (
+      root_path, "tests", "test_models", "data", "orange.raw", NULL);
+  ASSERT_TRUE (g_file_test (orange_raw_file, G_FILE_TEST_EXISTS));
+
+  fd = open (orange_raw_file, G_FILE_TEST_EXISTS);
```

Comment:
plz consider to get file content with g_file_get_content()

Reply:
Cool, the `g_file_get_contents` is way more safer. I updated the code with it, Thanks